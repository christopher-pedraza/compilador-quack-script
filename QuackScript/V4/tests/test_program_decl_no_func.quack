// Program with variable and constant declarations, no functions
program TestProgram;
main {
    print("Hola");
}
end