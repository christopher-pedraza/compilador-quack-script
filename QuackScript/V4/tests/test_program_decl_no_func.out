Parse Tree:
program_no_decl
  program
  id	TestProgram
  ;
  main
  body_statements
    {
    print_single
      print
      (
      cte_string	"Hola"
      )
      ;
    }
  end

IR:
('program', 'TestProgram', [], [], ('body_statements', [('print', [('cte_string', 'Hola')])]))

Symbol Table:
Container: global
  Parameters:
  Symbols:

